Chisel/CHLObjectAllocations.m

```diff
@@ -0,0 +1,49 @@
+//
+//  CHLObjectAllocations.m
+//  Chisel
+//
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#if __has_feature(objc_arc)
+#error CHLObjectAllocations.m expects ARC to be disabled.
+#endif
+
+#import "CHLAllocations.h"
+
+#import <objc/runtime.h>
+
+void CHLEnumerateObjectsWithBlock(void (^block)(id object))
+{
+  unsigned int classCount;
+  Class *classList = objc_copyClassList(&classCount);
+  NSPointerFunctionsOptions options = NSPointerFunctionsOpaqueMemory|NSPointerFunctionsOpaquePersonality;
+  NSHashTable *classes = [[NSHashTable alloc] initWithOptions:options capacity:classCount];
+  for (unsigned int i = 0; i < classCount; i++) {
+    [classes addObject:classList[i]];
+  }
+  free(classList);
+  
+  CHLEnumerateAllocationsWithBlock(^(vm_range_t range) {
+    if (range.size < sizeof(struct objc_object)) {
+      return;
+    }
+    
+    id object = (id)range.address;
+    Class class_ = object_getClass(object);
```

Comment:
I wonder what all possible outcomes are when `object` is not actually an objc object. At this point, all that is known is that the memory is large enough, but a lot of allocations will pass that test. Will `object_getClass` return `nil` if it's given something that's not an objc object, or could it assert and crash? If the latter, this code will need it's own `getClass` implementation.